Mingw64 compile problems with recommended options

$ g++ paq8pxd.cpp -DWINDOWS -DMT -msse2 -O3 -s -static -o paq8pxd.exe
paq8pxd.cpp:554:0: warning: "MT" redefined
 #define MT            //uncomment for multithreading, compression only

<command-line>:0:0: note: this is the location of the previous definition
C:\msys64\tmp\cclTlQgb.o:paq8pxd.cpp:(.text+0x17588): undefined reference to `inflateInit_'
C:\msys64\tmp\cclTlQgb.o:paq8pxd.cpp:(.text+0x17630): undefined reference to `deflateInit2_'
C:\msys64\tmp\cclTlQgb.o:paq8pxd.cpp:(.text+0x1787a): undefined reference to `deflate'
C:\msys64\tmp\cclTlQgb.o:paq8pxd.cpp:(.text+0x178f0): undefined reference to `inflate'
C:\msys64\tmp\cclTlQgb.o:paq8pxd.cpp:(.text+0x17bc5): undefined reference to `deflateEnd'
C:\msys64\tmp\cclTlQgb.o:paq8pxd.cpp:(.text+0x17be0): undefined reference to `inflateEnd'
C:\msys64\tmp\cclTlQgb.o:paq8pxd.cpp:(.text+0x17de5): undefined reference to `inflateInit_'
C:\msys64\tmp\cclTlQgb.o:paq8pxd.cpp:(.text+0x17e8c): undefined reference to `inflate'
C:\msys64\tmp\cclTlQgb.o:paq8pxd.cpp:(.text+0x17f6c): undefined reference to `deflateEnd'
C:\msys64\tmp\cclTlQgb.o:paq8pxd.cpp:(.text+0x17fc7): undefined reference to `inflateInit2_'
C:\msys64\tmp\cclTlQgb.o:paq8pxd.cpp:(.text+0x18022): undefined reference to `inflateInit2_'
C:\msys64\tmp\cclTlQgb.o:paq8pxd.cpp:(.text+0x182c2): undefined reference to `deflateInit2_'
C:\msys64\tmp\cclTlQgb.o:paq8pxd.cpp:(.text+0x183ad): undefined reference to `deflate'
C:\msys64\tmp\cclTlQgb.o:paq8pxd.cpp:(.text+0x184a4): undefined reference to `deflateEnd'
C:\msys64\tmp\cclTlQgb.o:paq8pxd.cpp:(.text+0x2845a): undefined reference to `inflate'
C:\msys64\tmp\cclTlQgb.o:paq8pxd.cpp:(.text+0x28469): undefined reference to `inflateEnd'
C:\msys64\tmp\cclTlQgb.o:paq8pxd.cpp:(.text+0x2b332): undefined reference to `inflate'
C:\msys64\tmp\cclTlQgb.o:paq8pxd.cpp:(.text+0x2b376): undefined reference to `inflateEnd'
C:\msys64\tmp\cclTlQgb.o:paq8pxd.cpp:(.text+0xc773): undefined reference to `inflateInit_'
C:\msys64\tmp\cclTlQgb.o:paq8pxd.cpp:(.text+0xc793): undefined reference to `inflateInit2_'
collect2.exe: error: ld returned 1 exit status

P.S. just need -lz into command line

but we have warning below:

 g++ paq8pxd.cpp -DWINDOWS -DMT -msse2 -O3 -s -static -o paq8pxd.exe -lz
paq8pxd.cpp:554:0: warning: "MT" redefined
 #define MT            //uncomment for multithreading, compression only

<command-line>:0:0: note: this is the location of the previous definition

--- a/README.md
+++ b/README.md
@@ -87,18 +87,18 @@ Recommended compiler commands and optimizations:
 
   MINGW g++ (x86,x64):
    with multithreading:
-    g++ paq8pxd.cpp -DWINDOWS -DMT -msse2 -O3 -s -static -o paq8pxd.exe 
+    g++ paq8pxd.cpp -DWINDOWS -DMT -msse2 -O3 -s -static -lz -o paq8pxd.exe 
    without multithreading:
-    g++ paq8pxd.cpp -DWINDOWS -msse2 -O3 -s -static -o paq8pxd.exe 
+    g++ paq8pxd.cpp -DWINDOWS -msse2 -O3 -s -static -lz -o paq8pxd.exe 
 
   UNIX/Linux (PC x86,x64):
    with multithreading:
-    g++ paq8pxd.cpp -DUNIX -DMT -msse2 -O3 -s -static -lpthread -o paq8pxd
+    g++ paq8pxd.cpp -DUNIX -DMT -msse2 -O3 -s -static -lpthread -lz -o paq8pxd
    without multithreading:
-    g++ paq8pxd.cpp -DUNIX -msse2 -O3 -s -static -lpthread -o paq8pxd
+    g++ paq8pxd.cpp -DUNIX -msse2 -O3 -s -static -lpthread -lz -o paq8pxd
 
   Non PC (e.g. PowerPC under MacOS X)
-    g++ paq8pxd.cpp -O2 -DUNIX -s -o paq8pxd
+    g++ paq8pxd.cpp -O2 -DUNIX -s -lz -o paq8pxd
 
 
 # ARCHIVE FILE FORMAT

